addons/escoria-core/game/core-scripts/esc/esc_room_manager.gd

```diff
@@ -32,8 +32,146 @@ func register_reserved_globals() -> void:
 			RESERVED_GLOBALS[key])
 
 
+# Performs the actions needed in order to change the current scene to the one 
+# specified by room_path.
+#
+# #### Parameters
+#
+# * room_path: Node path to the room that is to become the new current room.
+# * enable_automatic_transitions: Whether to play the transition between rooms 
+#	automatically or to leave the responsibility to the developer.
+func change_scene(room_path: String, enable_automatic_transitions: bool) -> void:
+	# Clear the event queue to remove other events (there could be duplicate
+	# events in there so we avoid running these multiple times)
+	escoria.event_manager.clear_event_queue()
+	
+	var exited_previous_room = false
+	
+	# If auto transition is enabled, try to determine whether we just exited a 
+	# room previously, so that we must play the auto transition out or not.
+	# This must happen if ESC_LAST_SCENE is set, or if we're running an 
+	# exit_scene event. Also room selector actions require the transition.
+	if enable_automatic_transitions \
+			and (
```

Comment:
I'd put that in the line above so that all the following lines can be indented more to the left.